src/Datadog.Trace.ClrProfiler.Managed/Integrations/AdoNetIntegration.cs

```diff
@@ -118,16 +120,17 @@ public static object ExecuteDbDataReader(object @this, int behavior, int opCode,
             var instanceType = command.GetType();
             var commandBehavior = (CommandBehavior)behavior;
             var callingAssembly = Assembly.GetCallingAssembly();
-            var dataReaderType = callingAssembly.GetType(DbDataReader);
+            var instrumentedType = @this.GetInstrumentedType(DbCommand);
```

Comment:
I like the approach of finding the base type and I think we can even extend that to safely find interface types 👍